src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -292,6 +294,17 @@ private Task WriteFormattedInt32Async(int value, CancellationToken cancellationT
                     _connectionClose = true;
                 }
 
+                // Before creating the response stream, check to see if we're done sending any content,
+                // and propagate any exceptions that may have occurred.  The most common case is that
+                // the server won't send back response content until it's received the whole request,
+                // so the majority of the time this task will be complete.
+                Task sendRequestContentTask = _sendRequestContentTask;
```

Comment:
> The socket Send will pend and the sendRequestContentTask will not be complete, correct?

The size of the send doesn't matter, nor how many packets it takes to send it.  If I'm sending you a request, and you don't send me a response until you've received the full request, then by definition I know the moment I get any response from you that you've received the full request, and at this point in the code, we've already received the beginning of the response from you.  The comment is citing that case, saying the most common case is that the server won't send back a response until it's received the full request, and so the majority of the time at this point this task will be complete because we wouldn't be at this point in the code unless we've received the beginning of the response.